sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -3,27 +3,89 @@
 
 #pragma once
 
+#include <string>
+#include <vector>
+
 #include <internal/contract.hpp>
 
+using namespace std;
+
 namespace azure
 {
 namespace core
 {
 namespace http
 {
 
-class http_request
+namespace http_method
 {
+enum HttpMethod
+{
+  GET,
+  HEAD,
+  POST,
+  PUT,
+  DELETE,
+  PATCH,
+};
+}
 
+class Header
```

Comment:
BTW, can we get away with `std::pair<std::string, std::string>` insteadof the Header class? You can store them in a std::map, and that will take care of duplicates.

Reply:
yeah, I like it.
Wondering if having Header as a class would have any benefit like serializing or .toString() function that we can use.
Otherwise, yeah, a map sounds better

Reply:
Done